client/quantum_serverless/core/job.py

```diff
@@ -303,10 +303,28 @@ def logs(self) -> str:
         """Returns logs of the job."""
         return self._job_client.logs(self.job_id)
 
-    def result(self):
-        """Return results of the job."""
+    def result(self, wait = True, cadence = 5, verbose = False):
+        """Return results of the job.
+        Args:
+            wait: flag denoting whether to wait for the
+                job result to be populated before returning
+            verbose: flag denoting whether to log a heartbeat
+                while waiting for job result to populate
+            cadence: time to wait between checking if job has
+                been terminated
+        """
+        if wait:
+            while self._in_terminal_state():
+                time.sleep(cadence)
```

Comment:
@psschwei @IceKhan13 

Paul recommended using ray for the timeout here. I always default to using pure Python any time I can. I'm really indifferent. Any thoughts @IceKhan13 ? I guess I'd just want to be consistent with what is being done in the rest of the repo in these cases.

`time.sleep`, `ray.wait`, and `ray.get` brings up exactly one instance apiece in the Github search

Reply:
`wait`, `get` works on task references inside of the program (ray job). We are working from outside, so your approach is correct :) 
Just fyi, here is what we have from outside https://docs.ray.io/en/latest/cluster/running-applications/job-submission/jobs-package-ref.html#jobsubmissionclient